components/brave_extension/extension/brave_extension/background/webDiscoveryProject.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import { App } from '../web-discovery-project/build'
+
+if (App !== undefined) {
+  const APP = new App({
+    version: chrome.runtime.getManifest().version
+  })
+
+  const WEB_DISCOVERY_PREF_KEY = 'brave.web_discovery_enabled'
+
+  const toggleWebDiscovery = (pref: chrome.settingsPrivate.PrefObject) => {
+    if (pref && pref.type === chrome.settingsPrivate.PrefType.BOOLEAN) {
+      const enable = pref.value
+      if (enable) {
+        // enable
+        APP.start()
+          .then(
+            () => {
+              APP.prefs.set('modules.web-discovery-project.enabled', true)
+              APP.prefs.set('modules.hpnv2.enabled', true)
```

Comment:
Same remark as below regarding not having to manage the prefs at all from here. I will push a change in the package directly so that you can simplify this code.

Reply:
https://github.com/brave/humanweb/releases/tag/v1.1.5

You can use this release now, which has all modules enabled by default and you can use `start()` and `stop()` to manage the opt-in/out.